nexus/db-queries/src/db/datastore/affinity.rs

```diff
@@ -703,6 +842,129 @@ impl DataStore {
         Ok(())
     }
 
+    async fn anti_affinity_group_member_add_group(
+        &self,
+        opctx: &OpContext,
+        authz_anti_affinity_group: &authz::AntiAffinityGroup,
+        affinity_group_id: AffinityGroupUuid,
+    ) -> Result<(), Error> {
+        let err = OptionalError::new();
+        let conn = self.pool_connection_authorized(opctx).await?;
+        self.transaction_retry_wrapper("anti_affinity_group_member_add_group")
+            .transaction(&conn, |conn| {
+                let err = err.clone();
+                use db::schema::anti_affinity_group::dsl as anti_affinity_group_dsl;
+                use db::schema::affinity_group::dsl as affinity_group_dsl;
+                use db::schema::affinity_group_instance_membership::dsl as a_instance_membership_dsl;
+                use db::schema::anti_affinity_group_affinity_membership::dsl as aa_affinity_membership_dsl;
+                use db::schema::sled_resource_vmm::dsl as resource_dsl;
+
+                async move {
+                    // Check that the anti-affinity group exists
+                    anti_affinity_group_dsl::anti_affinity_group
+                        .filter(anti_affinity_group_dsl::time_deleted.is_null())
+                        .filter(anti_affinity_group_dsl::id.eq(authz_anti_affinity_group.id()))
+                        .select(anti_affinity_group_dsl::id)
+                        .first_async::<uuid::Uuid>(&conn)
```

Comment:
Could this instead be implemented using the existing `DatastoreCollection` machinery, which IIUC is mostly for dealing with this sort of thing?